engine/utils/parseFilters.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// FilterMapByStructFields returns a new map containing only the fields that exist in the provided struct.
+func FilterMapByStructFields(fieldMap map[string]interface{}, filtersStruct interface{}) (map[string]interface{}, error) {
```

Comment:
I have few concerns about this function:
1. It can't handle date ranges and custom flags like "IncludeDeleted"
2. It has the same issue as with your previous PR, that actually gorm names are different then json names (and actually in the future if we would like to switch to camelcase on json it would make it even more different)
3. there are no tests that would show what is the expected behaviour of this function and if it is really doing what it meant to do.


So from my point of view I would go with something more straight forward like something what @chris-4chain suggested in previous PR. So something like:
```
conditions := destFilters.ToConditions()

// ....

func (d *DestinationFilters) ToConditions() {

if d.LockingScript != nil {
  addToConditions(conditions, "locking_script", d.LockingScript)
}

//...

```